src/NodeServices/WorkspaceEvents.cs

```diff
@@ -19,6 +20,17 @@ public static void OnWorkspaceAdded(Guid id, string name)
         }
 
         /// <summary>
+        /// An event that is triggered when a workspace is being
+        /// removed from the DynamoModel's workspaces collection.
+        /// </summary>
+        public static event WorkspaceRemoveStartedEventHandler WorkspaceRemoveStarted;
+        public static void OnWorkspaceRemoveStarted(Guid id, string name)
+        {
+            if (WorkspaceRemoveStarted != null)
```

Comment:
@ke-yu Can you also clarify in the comments when this event will actually be called? Something like:
```
 An event that is triggered prior to the removal of the workspace from the Workspaces collection.
```